AntennaTracker/mode_auto.cpp

```diff
@@ -4,9 +4,48 @@
 
 void ModeAuto::update()
 {
+    State old_state = _state;
+
     if (tracker.vehicle.location_valid) {
         update_auto();
-    } else if (tracker.target_set || (tracker.g.auto_opts.get() & (1 << 0)) != 0) {
+        _state = target_range_acceptable() ? State::TRACKING : State::SUPPRESSED;
+    } else if (tracker.target_set || scan_for_any_target()) {
         update_scan();
+        _state = State::SCANNING;
+    } else {
+        _state = State::IDLE;
+    }
+
+    // Announce the state change if necessary
+    if (old_state != _state) {
+        switch (_state) {
+            case State::TRACKING:
+                gcs().send_text(MAV_SEVERITY_INFO, "Tracker locked on the vehicle %d", (short)tracker.g.sysid_target);
+                break;
+
+            case State::SCANNING:
+                if (scan_for_any_target()) {
+                    gcs().send_text(MAV_SEVERITY_INFO, "Tracker scanning for a vehicle");
+                } else {
+                    gcs().send_text(MAV_SEVERITY_INFO, "Tracker scanning for the vehicle %d", (short)tracker.g.sysid_target);
+                }
+                break;
+
+            case State::SUPPRESSED:
+                gcs().send_text(MAV_SEVERITY_INFO, "Tracker suppressed while the target is within %d m", (short)tracker.g.distance_min);
```

Comment:
```suggestion
                gcs().send_text(MAV_SEVERITY_INFO, "Tracker suppressed while the target is within %i m", tracker.g.distance_min);
```

Reply:
I would have expected this to work, but:
```
../../AntennaTracker/mode_auto.cpp: In member function ‘virtual void ModeAuto::update()’:
../../AntennaTracker/mode_auto.cpp:23:84: error: format ‘%i’ expects argument of type ‘int’, but argument 4 has type ‘AP_Int16’ {aka ‘AP_ParamT<short int, AP_PARAM_INT16>’} [-Werror=format=]
   23 |                 gcs().send_text(MAV_SEVERITY_INFO, "Tracker locked on the vehicle %i", tracker.g.sysid_target);
      |                                                                                   ~^   ~~~~~~~~~~~~~~~~~~~~~~
      |                                                                                    |             |
      |                                                                                    int           AP_Int16 {aka AP_ParamT<short int, AP_PARAM_INT16>}
```

Compiler: `gcc (Gentoo 13.2.1_p20230826 p7) 13.2.1 20230826`
Clang (LSP client) warns about the same. I tried `%hi` as well, but for some reason it's also treated as `int`?

Reply:
Ah, you could try adding the `get()`. That makes the AP_Param type return the underlying type.
```suggestion
                gcs().send_text(MAV_SEVERITY_INFO, "Tracker suppressed while the target is within %i m", tracker.g.distance_min.get());
```